### PR TITLE
feat(@lexical/devtools): Added TreeView rendering instead of a simple textarea

### PR DESCRIPTION
**Pull request by StyleT**
List of changes:
- Added `@lexical/devtools-core` that will contain all UIs for Lexical debugging. As we may want to use them outside of the extension (like now `TreeView` is a part of the `@lexical/react`)
- Added TreeView rendering instead of a simple textarea within extension
- Module resolution config for Vite now is shared between playground and extension
- Exported `EditorSetOptions` type from `lexical` package

**Feature demo in Chrome:**

https://github.com/facebook/lexical/assets/1125900/6eaa90ba-eb13-408f-9ba8-d10c8aeeebd7

**Feature demo in Firefox:**

https://github.com/facebook/lexical/assets/1125900/f3b9e9da-9485-47df-ab88-314fd4c84e4c

### CODE REVIEW

File: packages/lexical-devtools/src/entrypoints/injected/main.ts [hunk 1]
```diff
@@ -6,60 +6,60 @@
...
-  setStatesForTab(
-    tabID,
-    Object.fromEntries(editors.map((e) => [e._key, e.getEditorState()])),
+  onMessage('refreshLexicalEditorsForTabID', () =>
+    scanAndListenForEditors(tabID, extensionStore, commandLog),
   );
+  onMessage('generateTreeViewContent', (message) => {
+    const editor = queryLexicalEditorByKey(message.data.key);
+    if (editor == null) {
+      throw new Error(`Can't find editor with key: ${message.data.key}`);
```
Comment (fantactuka) [inline packages/lexical-devtools/src/entrypoints/injected/main.ts:36]: Q: what happens when extension throws?

Comment (StyleT) [reply packages/lexical-devtools/src/entrypoints/injected/main.ts:36]: Error will be passed to the caller side (devtools tab) and there we show message to the used at the top of the page now.

It's lame, so better error/log system shall be in place, but not in this PR :P I'll create issue for this

File: packages/lexical-devtools-core/src/useLexicalCommandsLog.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            if (newState.length > 10) {
+              newState.shift();
+            }
+
+            return newState;
+          });
+
+          return false;
+        },
+        COMMAND_PRIORITY_HIGH,
```
[the remark this answers is not in the document]
Comment (StyleT) [reply packages/lexical-devtools-core/src/useLexicalCommandsLog.ts:47]: - I think we shall add your 2nd concern to the refresh procedure (I'll create issue)

Regarding criticality - I just moved this code from existing TreeView, but your point sounds right, will update it

File: packages/lexical-devtools-core/src/useLexicalCommandsLog.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  for (const [command] of editor._commands) {
+    unregisterCommandListeners.add(
+      editor.registerCommand(
+        command,
+        (payload) => {
+          setLoggedCommands((state) => {
+            const newState = [...state];
+            newState.push({
+              payload,
+              type: command.type ? command.type : 'UNKNOWN',
```
[the remark this answers is not in the document]
Comment (StyleT) [reply packages/lexical-devtools-core/src/useLexicalCommandsLog.ts:35]: This change causes error in WWW build, so leaving as is:
```
node scripts/build.js --www --prod
/home/runner/work/lexical/lexical/node_modules/@ampproject/rollup-plugin-closure-compiler/dist/index.js:1571
[33](https://github.com/facebook/lexical/actions/runs/8604009341/job/23577172760?pr=5830#step:9:34)
                reject(new Error(`Google Closure Compiler exit ${exitCode}: ${stdErr}`));

ERROR - [JSC_LANGUAGE_FEATURE] This language feature is only supported for ECMASCRIPT_***0***0 mode or better: Nullish coalescing.
```

File: packages/lexical-devtools-core/src/useLexicalCommandsLog.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (StyleT) [reply packages/lexical-devtools-core/src/useLexicalCommandsLog.ts]: Updated!

File: packages/lexical-devtools/src/entrypoints/injected/main.ts (hunk 1 above)
Comment (StyleT) [reply packages/lexical-devtools/src/entrypoints/injected/main.ts:36]: Added to the plan here: https://github.com/facebook/lexical/issues/5675#issuecomment-2043312215

**Review by zurfyx — APPROVED** [review]

**Pull request merged**